internal/handlers/pg/handler.go

```diff
@@ -50,166 +43,20 @@ type Handler struct {
-		msg := reqBody.(*wire.OpMsg)
-		var document *types.Document
-		document, err = msg.Document()
-		command = document.Command()
-		requests.WithLabelValues(command).Inc()
-
-		if err == nil {
-			resHeader.OpCode = wire.OP_MSG
-			resBody, err = h.handleOpMsg(ctx, msg, command)
-		}
-
-	case wire.OP_QUERY:
-		query := reqBody.(*wire.OpQuery)
-		command = query.Query.Command()
-		requests.WithLabelValues(command).Inc()
-
-		resHeader.OpCode = wire.OP_REPLY
-		resBody, err = h.CmdQuery(ctx, query)
-
-	case wire.OP_REPLY:
-		fallthrough
-	case wire.OP_UPDATE:
-		fallthrough
-	case wire.OP_INSERT:
-		fallthrough
-	case wire.OP_GET_BY_OID:
-		fallthrough
-	case wire.OP_GET_MORE:
-		fallthrough
-	case wire.OP_DELETE:
-		fallthrough
-	case wire.OP_KILL_CURSORS:
-		fallthrough
-	case wire.OP_COMPRESSED:
-		fallthrough
-	default:
-		requests.WithLabelValues(command).Inc()
-
-		err = lazyerrors.Errorf("unexpected OpCode %s", reqHeader.OpCode)
-	}
-
-	// set body for error
-	if err != nil {
-		switch resHeader.OpCode {
-		case wire.OP_MSG:
-			protoErr, recoverable := common.ProtocolError(err)
-			closeConn = !recoverable
-			var res wire.OpMsg
-			err = res.SetSections(wire.OpMsgSection{
-				Documents: []*types.Document{protoErr.Document()},
-			})
-			if err != nil {
-				panic(err)
-			}
-			resBody = &res
-			result = pointer.ToString(protoErr.Code().String())
-
-		case wire.OP_QUERY:
-			fallthrough
-		case wire.OP_REPLY:
-			fallthrough
-		case wire.OP_UPDATE:
-			fallthrough
-		case wire.OP_INSERT:
-			fallthrough
-		case wire.OP_GET_BY_OID:
-			fallthrough
-		case wire.OP_GET_MORE:
-			fallthrough
-		case wire.OP_DELETE:
-			fallthrough
-		case wire.OP_KILL_CURSORS:
-			fallthrough
-		case wire.OP_COMPRESSED:
-			fallthrough
-
-		default:
-			// do not panic to make fuzzing easier
-			h.l.Error("Handler error for unexpected response opcode", zap.Error(err), zap.Stringer("opcode", resHeader.OpCode))
-			closeConn = true
-			result = pointer.ToString("unexpected")
-			return
-		}
-	}
-
-	// TODO Don't call MarshalBinary there. Fix header in the caller?
-	// https://github.com/FerretDB/FerretDB/issues/273
-	b, err := resBody.MarshalBinary()
-	if err != nil {
-		result = nil
-		panic(err)
-	}
-	resHeader.MessageLength = int32(wire.MsgHeaderLen + len(b))
-
-	resHeader.RequestID = atomic.AddInt32(&h.lastRequestID, 1)
-	resHeader.ResponseTo = reqHeader.RequestID
-
-	if result == nil {
-		result = pointer.ToString("ok")
-	}
-
-	return
-}
-
-func (h *Handler) handleOpMsg(ctx context.Context, msg *wire.OpMsg, cmd string) (*wire.OpMsg, error) {
-	if cmd, ok := common.Commands[cmd]; ok {
-		if cmd.Handler != nil {
-			return cmd.Handler(h, ctx, msg)
-		}
-	}
-
-	errMsg := fmt.Sprintf("no such command: '%s'", cmd)
-	return nil, common.NewErrorMsg(common.ErrCommandNotFound, errMsg)
-}
-
 // MsgDebugError used for debugging purposes.
 func (h *Handler) MsgDebugError(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
```

Comment:
There is no way to call those methods now.

Reply:
added.